core/Rename.cpp

```diff
@@ -93,10 +114,49 @@ namespace olympia
 
         credits_dispatch_ += credits;
         if (uop_queue_.size() > 0) {
-            ev_rename_insts_.schedule();
+            ev_schedule_rename_.schedule();
         }
     }
+    void Rename::getAckFromROB_(const InstPtr & inst_ptr)
+    {
+        sparta_assert(inst_ptr->getStatus() == Inst::Status::RETIRED,
+                        "Get ROB Ack, but the inst hasn't retired yet!");
+
+        core_types::RegFile rf = core_types::RegFile::RF_INTEGER;
+        auto const & dests = inst_ptr->getDestOpInfoList();
+        auto const & dest = inst_ptr->getRenameData().getDestination();
+        auto const & original_dest = inst_ptr->getRenameData().getOriginalDestination();
+        if(dests.size() > 0){
+            sparta_assert(dests.size() == 1); // we should only have one destination
+            rf = determineRegisterFile(dests[0]);   
+            // free PRF if no references from srcs
+            if(reference_counter_[rf][dest] <= 0){
+                freelist_[rf].push(dest);
+            }
+            map_table_[rf][original_dest].first = 0;
+            // map entry is now invalid, new srcs should read from ARF
+            map_table_[rf][original_dest].second = false;
+        }
 
+        auto const & srcs = inst_ptr->getRenameData().getSource();
+        // freeing references to PRF
+        for(auto src: srcs){
```

Comment:
Consider `const auto &` to avoid copy

Reply:
Addressed in [1bf68f4](https://github.com/riscv-software-src/riscv-perf-model/pull/40/commits/1bf68f4f79753f81792e51ef39c2b06e261e82de)